cli/compose/types/types.go

```diff
@@ -191,6 +193,15 @@ type ServiceNetworkConfig struct {
 	Ipv6Address string `mapstructure:"ipv6_address"`
 }
 
+// ServiceSecretConfig is the secret configuration for a service
+type ServiceSecretConfig struct {
```

Comment:
We have lots of different types.
* public API types
* internal types
* config parsing types

Everything in `cli/compose/types` is used to define the structure of the Compose file format. We can't use the API types, because the format doesn't map perfectly to the API. There are always going to be slight variations between the types, and different trade-offs because of how they are used. There are also variations because of ID fields (that get provided by swarm, not from a config).